receiver/prometheusreceiver/internal/metadata.go

```diff
@@ -45,7 +45,7 @@ func (t *mService) Get(job, instance string) (MetadataCache, error) {
 
 	// from the same targetGroup, instance is not going to be duplicated
 	for _, target := range targetGroup {
-		if target.DiscoveredLabels().Get(model.AddressLabel) == instance {
+		if target.Labels().Get(model.InstanceLabel) == instance {
```

Comment:
Would it be hard to add a test reproducing the original issue (and succeeding after this change)?
Probably something invoking
https://github.com/open-telemetry/opentelemetry-collector/blob/c0dd66ef47bd94e7578a874583903059a2fe7925/receiver/prometheusreceiver/internal/transaction.go#L123-L128

Reply:
+1.

Reply:
What kind of test you mean here? Simple unit test validating that we pick up correct labels after renaming or e2e test running service discovery?

Reply:
I think we can merge this change without new tests.

Unit tests for transaction in `transaction_test.go` are using a mock `MetadataService`, so you won't be able to test the change there. We could replace it with a real `mService` which internally uses a mock `scrape.Manager` for better test coverage, but seems too much work for one-line change.

AFAICT, there are no e2e tests for prometheus receiver currently, only OTLP and OpenCensus receivers are covered in https://github.com/open-telemetry/opentelemetry-collector/blob/master/testbed/tests/metric_test.go
Would be nice to add one, but that's also out of scope of this PR.